reactive/containerd.py

```diff
@@ -248,6 +269,50 @@ def merge_custom_registries(config_directory, custom_registries,
     return registries
 
 
+def invalid_custom_registries(custom_registries):
+    """
+    Validate custom registries from config.
+
+    :param str custom_registries: juju config for custom registries
+    :return: error string for blocked status if condition exists, None otherwise
+    :rtype: Optional[str]
+    """
+    try:
+        registries = _registries_list(custom_registries)
+    except json.JSONDecodeError:
+        return "Failed to decode json string"
+    except ValueError as e:
+        return "custom_registries is not a list"
+
+    required_fields = ['url']
+    str_fields = ['host', 'username', 'password', 'ca_file', 'cert_file', 'key_file']
+    truthy_fields = ['insecure_skip_verify', ]
+    restricted_fields = ['ca', 'cert', 'key']
+    host_set = set()
+    for idx, registry in enumerate(registries):
+        if not isinstance(registry, dict):
+            return "registry #{} is not in object form".format(idx)
+        for field in required_fields:
+            if field not in registry:
+                return "registry #{} missing required field {}".format(idx, field)
+        for field in required_fields + str_fields:
+            value = registry.get(field)
+            if value and not isinstance(value, str):
+                return "registry #{} field {}={} is not a string".format(idx, field, value)
+        for field in truthy_fields:
+            value = registry.get(field)
+            if isinstance(value, str) and "false" in value.lower():
+                return "registry #{} field {}='{}' is not falsey".format(idx, field, value)
```

Comment:
This is a bit weird. I could have insecure_skip_verify="no" and it would be accepted as valid and truthy.

I would go further and say that insecure_skip_verify should always be a boolean, never a string.

Reply:
I was concerned about current deployments where `insecure_skip_verify` may be something like "true" rather than True.  But I guess the point of this is to block letting the customer know they've got invalid config. I'll update the README as well